core/meta/src/TSchemaRuleSet.cxx

```diff
@@ -115,7 +115,8 @@ Bool_t TSchemaRuleSet::AddRule( TSchemaRule* rule, EConsistencyCheck checkConsis
      R__LOCKGUARD(gInterpreterMutex);
      streamerInfosTest = (fClass->GetStreamerInfos()==nullptr || fClass->GetStreamerInfos()->IsEmpty());
    }
-   if( rule->GetTarget()  && !(fClass->TestBit(TClass::kIsEmulation) && streamerInfosTest) ) {
+   if (rule->GetTarget()  && fClass->GetState() > TClass::kEmulated && streamerInfosTest)
```

Comment:
Would perhaps a comment about the meaning of a state greater than kEmulated in this context?

Reply:
I could but the answer should be straight-forward from the declaration of the `EState`: https://root.cern.ch/doc/master/TClass_8h_source.html#l00122 (I.e. we have "more" information that we would have from Emulated and lower) and we use this pattern in a few places.